### DISCUSSION TITLE
Multiple Instances

### DISCUSSION THREAD

**Discussion by Jenesius** [Jenesius/vue-modal#94] [Q&A] [answered]
Main idea: Add levels (namespace) for displaying modal windows. This approach can help unify this library. For example, I thought a lot about displaying notifications, like in the following screenshot:

![image](https://github.com/Jenesius/vue-modal/assets/21982550/7eee116e-5a80-4ea6-8281-cd58266b8aa6)

To some extent, this is also a set of components that are shown in a **separate* layout, but with a slightly different specification, a slightly different display. But this can be changed more easily via `css`.

As a result:
- develop the `namespace` mechanism
- develop a mechanism for mulltiple containers
- tests for window display namespace and multiple containes
- documentation

**Comment by Jenesius** [comment]
```vue
<template>
  <container namespace = "notification"/>
  <container /> <!--Default container-->
</template>
<script setup>
import {container} from "jenesius-vue-modal"

</script>
```
For container i think its more simple solution.
If we say about push, prompt, open - its more hard.
```ts
namespace("notification").pushModal(modal)
pushModal(Modal)
```
Or 
```ts
pushModal(modal, {}, { namespace: "notification" }) // Looks more better
```
One of the options I suggested.

**Comment by Jenesius** [comment]
Also we need to create analog for modalQueue.
```ts
const queue = getQueueByNamespace("notification"); // reactive []
```
So, in result we can create our own layout.
Also in addModal this line should be updated:
From:
```ts
modalQueue.value.push(markRaw(modal));
```
To:
```ts
const modalQueue = getQueueByNamespace(options.namespace):
modalQueue.value.push(markRaw(modal));
```

**Comment by Jenesius** [comment]
Don't forgot about getCurrentModal. We can pass namespace name as second param.

**Comment by anshirko** [comment, marked as answer]
As a result:
- implement a method for receiving a queue via namespace
- add namespace option for AddModalOptions
- add option for ModalContainer, `namespace: string`
- separate tests for namespace
- documentation for namespace
- demo for notification based on namespace

**Comment by Jenesius** [comment]
I believe that on the basis of this task it is possible to implement adequate storage of modal windows. Also in the modal window you need to specify the namespace to which it belongs.

**Comment by Jenesius** [comment]
![image](https://github.com/Jenesius/vue-modal/assets/21982550/8ced0ec2-ab3c-44a0-9f4d-e94272da0081)
The error blocks is also ModalInstance but with namespace: "notification".
So, now we can create our own container-queue.

**Comment by Jenesius** [comment]
Now there is a problem with storing modal windows. We store in both modalQueue and Store. It is not right. Removing a modal window from storage should automatically remove it from the queue.

**Comment by Jenesius** [comment]
I think that for this particular task you can change `ref` -> `reactive`. Or implement a singly linked list in which modal windows will be stored.
